test/unit/rtc_session.js

```diff
@@ -760,4 +760,52 @@ describe('RTC session', () => {
             chai.expect(session._onSessionDestroyed.calledOnce).to.be.true;
         });
     });
+
+    describe('getStats', () => {
+        /**
+         * @type {RtcSession}
+         */
+        var session;
+        var clock;
+
+        beforeEach(() => {
+            const dependencyModule = require("../../src/js/rtp-stats");
+            sinon.stub(dependencyModule, "extractMediaStatsFromStats").returns(null);
+            session = new RtcSession('wss://amazon-connect-rtc-server.amazonaws.com/', [], 'contactToken', console);
+            session._legacyStatsReportSupport = true;
+            var rawStats = {
+                result: () => {return null}
+            }
+            session._pc = {
+                signalingState: 'stable',
+                getStats:  (callback) => {setTimeout(function() {
+                    callback(rawStats);
+                  }, 1)}
+            };
+            var audioTrack = {
+                enabled: true
+            }
+            var audioTracks = [ audioTrack ];
+            session._remoteAudioStream = {
+                getAudioTracks: () => {return audioTracks;}
+            }
+            clock = sinon.useFakeTimers();
+        });
+
+        afterEach(() => {
+            clock.restore();
+        });
+
+        it('for legacy stats, when digested stats is empty, error is handled', () => {
+            var errorCatched = false;
+            session.getRemoteAudioStats().then(function () {
```

Comment:
This unit test is to make sure the callback in session._pc.getStats() will not throw an error(old behavior).
See detailed explanation about the bug here: https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw/33446005#33446005

getRemoteAudioStats will invoke, `async getStats() ` and call session._pc.getStats(). session._pc.getStats(). is mocked method that will just invoke the callback function after a delay. This means that the unit test will invoke the callback function and verified that the call back function no longer throw the error